lib/algoliasearch-rails.rb

```diff
@@ -773,7 +773,7 @@ def algolia_remove_from_index!(synchronous = false)
     end
 
     def algolia_enqueue_remove_from_index!(synchronous)
-      if algoliasearch_options[:enqueue]
+      if algoliasearch_options[:enqueue] && !@algolia_without_auto_index_scope
         algoliasearch_options[:enqueue].call(self, true)
       else
         algolia_remove_from_index!(synchronous)
```

Comment:
The first line of [`algolia_remove_from_index!`](https://github.com/algolia/algoliasearch-rails/blob/e0f086766a1768c8f5f889ee40a2faf67306b160/lib/algoliasearch-rails.rb#L499) is to check `@algolia_without_auto_index_scope` to see if `@algolia_without_auto_index_scope` is `true`, and return immediately if so.

Without the above change, if a queue is being used in conjunction with `without_auto_index`, then `@algolia_without_auto_index_scope` might go from `true` to `false` before the queued method runs and reaches the check inside `algolia_remove_from_index!`.

For the reference, here's what happens when you use `without_auto_index`:

```ruby
def algolia_without_auto_index(&block)
  @algolia_without_auto_index_scope = true
  begin
    yield
  ensure
    @algolia_without_auto_index_scope = false
  end
end
```

To avoid this race condition, I think we ought to check `@algolia_without_auto_index_scope` before dispatching a queuing method.